src/flexmeasures_client/client.py

```diff
@@ -85,6 +86,15 @@ def __post_init__(self):
             )
         if len(self.password) < 1:
             raise EmptyPasswordError("password cannot be empty")
+        self.separate_host_and_port()
+
+    def separate_host_and_port(self):
```

Comment:
Alright, *actually*  this function is rather something like `determine_port()`, right?

It sets the `port`, according to scheme, or otherwise from it being included in the host.

And that brings me to the following problem: `port` is a class variable, so it can be set by the user directly. 

But right now, the port will never be set that way. So this is possible:

```
>>> client = FlexMeasuresClient(email="nicolas@seita.nl", password="xxxx", host="127.0.0.1:5000", port=5001, ssl=False)
>>> client
FlexMeasuresClient(password='xxxx', email='nicolas@seita.nl', host='127.0.0.1', port=5000, ssl=False, api_version='v3_0', path='/api/v3_0/', access_token=None, max_polling_steps=10, polling_timeout=200.0, request_timeout=20.0, polling_interval=10.0, session=None)
```

Or:

```
>>> client = FlexMeasuresClient(email="nicolas@seita.nl", password="xxxx", host="127.0.0.1", port=5001, ssl=False)
>>> client
FlexMeasuresClient(password='xxxx', email='nicolas@seita.nl', host='127.0.0.1', port=80, ssl=False, api_version='v3_0', path='/api/v3_0/', access_token=None, max_polling_steps=10, polling_timeout=200.0, request_timeout=20.0, polling_interval=10.0, session=None)
```

We should probably
- never overwrite the port set by the user in this way
- throw an error if the user set a port explicitly but there is also a different port within the host string.